docs/dev/releases.md

```diff
@@ -0,0 +1,102 @@
+# Releases
+
+This doc explains how to build, test and release ingress controllers.
+
+## Building
+
+All ingress controllers are build through a Makefile. Depending on your
+requirements you can build a raw server binary, a local container image,
+or push an image to a remote repository.
+
+Build a raw server binary
+```console
+$ make controller
+```
+
+Build a local container image
+```console
+$ make container
+```
+
+Push the container image to a remote repository
+```console
+$ make push TAG=0.0 PREFIX=$USER/ingress-controller
+```
+
+## Dependencies
+
+The build should use dependencies in the `ingress/vendor` directory.
+Occasionally, you might need to update the dependencies.
+
+```console
+$ godep version
+godep v74 (linux/amd64/go1.6.1)
+$ go version
+go version go1.6.1 linux/amd64
+```
+
+This will automatically save godeps to `vendor/`
+```console
+$ godep save ./...
+```
+
+If you have an older version of `godep`
+```console
+$ go get github.com/tools/godep
+$ cd $GOPATH/src/github.com/tools/godep
+$ go build -o godep *.go
+```
+
+If you have an older version of `go`
```

Comment:
imo we should just assume the dev is using a new enough version of go and not provide instructions for older versions.